src/QGLView.cc

```diff
@@ -336,6 +337,21 @@ void QGLView::zoom(double v, bool relative)
     updateGL();
 }
 
+void QGLView::zoomCursor(int x, int y, int zoom)
+{
+  auto old_dist = cam.zoomValue();
+  this->cam.zoom(zoom, true);
+  auto dist = cam.zoomValue();
+  auto ratio = old_dist / dist - 1.0; 
+  // screen coordinates from -1 to 1
+  auto screen_x = 2.0 * (x + 0.5) / this->cam.pixel_width - 1.0;
+  auto screen_y = 1.0 - 2.0 * (y + 0.5) / this->cam.pixel_height;
+  auto height = dist * tan(cam.fov /2 * M_PI / 180);
+  auto mx = ratio*screen_x*(aspectratio*height);
+  auto mz = ratio*screen_y*height;
```

Comment:
You can const all the local variables in this function. Good for readability as you're promising not to change the values.